File: CarRentalClient/src/app/shared/models/Car.Model.ts
import { BranchModel } from "./Branch.model";
import { CartypeModel } from "./Cartype.model";

export interface CarModel {

  CarID: number;
  LicenseNumber: string;
  CurrentKilometers: number;
  IsOperative: boolean;
  BranchLocation: BranchModel;
  CarInfo: CartypeModel;
  CarPic: string;
  IsDeleted: boolean;
  
}

File: CarRentalServer/02_BOL/CarModel.cs
using System.ComponentModel.DataAnnotations;


namespace _02_BOL
{
    public class CarModel
    {

        public int CarID { get; set; }

        //public int CarTypeID { get; set; } //its not needed. I have Carinfo


        public string LicenseNumber { get; set; }


        public int CurrentKilometers { get; set; }


        public bool IsOperative { get; set; }

        [Required]
        public BranchModel BranchLocation { get; set; }

        [Required]
        public CarTypeModel CarInfo { get; set; }

        public string CarPic { get; set; }

        public bool IsDeleted { get; set; }
    }
}
